python/deltalake/__init__.py

```diff
@@ -23,7 +24,14 @@ def load_version(self, version: int) -> None:
         self._table.load_version(version)
 
     def to_pyarrow_dataset(self) -> pyarrow.dataset.Dataset:
-        return dataset(self._table.file_paths(), format="parquet")
+        paths = [urlparse(curr_file) for curr_file in self._table.file_paths()]
+
+        # Decide based on the first file, if the file is on cloud storage or local
+        if paths[0].netloc:
+            keys = [curr_file.path for curr_file in paths]
+            return dataset(keys, filesystem=paths[0].scheme + "://" + paths[0].netloc)
```

Comment:
Good practice to avoid string concatenation using `+`

```suggestion
            return dataset(keys, filesystem=f"{paths[0].scheme}://{paths[0].netloc}")
```